pkg/vm/ch.go

```diff
@@ -59,7 +62,7 @@ func (m *Machine) Run(ctx context.Context, socket, logs string) error {
 		args["--fs"] = filesystems
 	}
 
-	if len(m.Environment) > 0 {
+	if len(m.Environment) > 0 || m.Entrypoint != "" {
```

Comment:
It's not clear here why u suddenly checking on the m.Entrypoint as well. (I know you do this because appendEnv also append the entrypoint args). But this is not clear from the current method name.

Instead `appendEnv` should become `buildZosRC` or something. to make it clear that it does more than just Env.

Reply:
The check for `virtiofs` can also be moved to this method. so in Run you only call `buildZosRC` (or whatever) that checks if it actually has env or entrypoint. If no, return success. Otherwise find the root virtiofs (fail if can't find) then build the file.